### PR TITLE
Remove Grad Scaler + Switch To Llama1b

### PR DESCRIPTION
**Pull request by KMFODA**
This PR does the following:
- [x] Removes Gradient Scaler
- [x] Switches to Llama-1b
- [x] Adds gradient normalisation before `self.state_averager.step()`

### CODE REVIEW

File: distributed_training/validator/reward.py [hunk 1]
```diff
@@ -284,12 +240,12 @@ async def score_uid(self, uid: int):
 
                     with torch.autocast(device_type="cuda", dtype=torch.bfloat16):
                         outputs = self.model(input_ids=inputs, labels=labels)
-                        loss = outputs[1] / self.number_of_local_steps
+                        loss = outputs[0] / self.number_of_local_steps
```
[the remark this answers is not in the document]
Comment (KMFODA) [reply distributed_training/validator/reward.py]: Yes good suggestion

File: distributed_training/validator/reward.py [hunk 2]
```diff
@@ -262,7 +218,7 @@ async def score_uid(self, uid: int):
         )
 
         model_final = AutoModelForCausalLM.from_pretrained(
-            model_huggingface_id, revision=commits[1].commit_id, trust_remote_code=True
+            model_huggingface_id, revision=commits[1].commit_id, trust_remote_code=False
```
[the remark this answers is not in the document]
Comment (KMFODA) [reply distributed_training/validator/reward.py]: Yes very good spot, resolved.

File: neurons/miner.py [hunk 3]
```diff
@@ -166,7 +166,9 @@ def _init_model_components(self):
         self._setup_training_params()
 
     def _init_tokenizer(self):
-        self.tokenizer = AutoTokenizer.from_pretrained("distilgpt2", use_fast=False)
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            self.config.neuron.global_model_name, use_fast=False
+        )
```
[the remark this answers is not in the document]
Comment (KMFODA) [reply neurons/miner.py:171]: Yes, I had tested this before and there were some outdated library issues but I can't seem to replicate them now so I'll go ahead and enable here.

File: distributed_training/validator/reward.py [hunk 4]
```diff
@@ -416,74 +413,86 @@ def benchmark_untested_uids(self):
...
+        self.uid_tracker[uid].get("train_score", 0.0) for uid in self.uid_tracker
+    ]
+    all_all_reduce_scores = [
+        self.uid_tracker[uid].get("all_reduce_score", 0.0) for uid in self.uid_tracker
+    ]
+    all_repo_valid_scores = [
+        self.uid_tracker[uid].get("repo_valid_score", 0.0) for uid in self.uid_tracker
+    ]
+
+    train_score_norm = (
```
Comment (KMFODA) [inline distributed_training/validator/reward.py:464]: @jundi69 FYI this change had a bug where the new `all_reduce_score_norm` variable was a list vs before it was a number since we ran .item() on the output of `np.linalg.norm()`. This then caused `self.uid_tracker[uid]["total_score"]` to be a list which then caused the validator to crash when running: https://github.com/KMFODA/DistributedTraining/blob/a3015d35694a853252cc245c10c1e81c6cc6c3d2/distributed_training/base/validator.py#L429.

This was fixed before Friday's merge but it would be good to test for these types of errors before pushing commits like these. Just running a non_master validator on this commit for 1 or 2 steps would have highlighted this error.
